server/lib/analytics/index.js

```diff
@@ -4,15 +4,52 @@
  */
 
 import superagent from 'superagent';
+import { v4 as uuid } from 'uuid';
+import { isUndefined, omit, assign, get } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import config from 'config';
 import { statsdTimingUrl } from '../../../client/lib/analytics/statsd';
-import { isUndefined, omit, assign } from 'lodash';
 const URL = require( 'url' );
 
+function getUserFromRequest( request ) {
+	// if user has a cookie, lets use that
+	const encodedUserCookie = get( request, 'cookies.wordpress_logged_in', null );
+
+	if ( encodedUserCookie ) {
+		try {
+			const userCookieParts = decodeURIComponent( encodedUserCookie ).split( '|' );
+
+			// We don't trust it, but for analytics this is enough
+			return {
+				_ul: userCookieParts[ 0 ],
+				_ui: parseInt( userCookieParts[ 1 ], 10 ),
+				_ut: 'wpcom:user_id',
+			};
+		} catch ( ex ) {
+			// ignore the error and let it fallback to anonymous below
+		}
+	}
+
+	// if user doesn't have a cookie, and we had the user identification passed to us on query params
+	// we'll use that, otherwise, we'll just use anonymous.
+	let userIdType = get( request, 'query._ut', 'anon' );
```

Comment:
does it matter that `anon` is set? Can't we leave it undefined?

Reply:
Not sure what you meant exactly, userIdType ( `_ut` ) is a required parameter we need to pass to tracks

Reply:
If `_ut` is not set to one of the whitelisted values, the event will be rejected.